### PR TITLE
Merging m4_integrate_main Branch into Main

### PR DESCRIPTION
**Pull request by addie-p**
This merge request incorporates all of the changes from Milestone 4 from the working "integrated" main branch (used to avoid breaking any code on the main branch) to the main branch. 

The following major features were integrated in this branch and will be merged into the main branch:
DB Integration for Saved Restaurants (https://github.com/addie-p/Team20-Project/issues/125)
DB Integration for Geolocation/Filters (https://github.com/addie-p/Team20-Project/issues/126)
Implement Login & Authentication (https://github.com/addie-p/Team20-Project/issues/158)
Backend Integration for Rating System (https://github.com/addie-p/Team20-Project/issues/160)
Server-Side Setup/Implementation for Restaurant Cards (https://github.com/addie-p/Team20-Project/issues/162)
Server and Client-Side Saved Restaurant Photos (https://github.com/addie-p/Team20-Project/issues/178)

All code has been finalized and tested/approved by reviewers prior to merging.

Issue: #146

### CODE REVIEW

File: team/.DS_Store
Comment (helenliuf) [inline team/.DS_Store:1]: can possibly remove

File: images.db
Comment (helenliuf) [inline images.db:1]: can possibly remove

File: .DS_Store
Comment (helenliuf) [inline .DS_Store:1]: can possibly remove

File: package-lock.json
Comment (helenliuf) [inline package-lock.json:1]: can possibly remove

**Review by helenliuf — APPROVED** [review]
looks good to merge though some files can be removed as not necessary or could cause problems for new users

**Review by jadatwo — APPROVED** [review]
Tested and web application functions correctly by integrating backend. User must install node modules. Will probably need to remove .DS_store files as it could create issues.

**Review by cding888 — APPROVED** [review]
pulled to my local repository and works great! good to merge!

**Review by aliyabedeen — APPROVED** [review]
Tested the code on my branch and everything looks great! looks good to merge.

**Review by evc29 — APPROVED** [review]
Tested on my end and everything looks great! We can possibly remove the .DS_store files but other than that, everything looks up to speed and all our components look great together! Ready to merge!

**Review by SandraV255 — APPROVED** [review]
Tested remotely and all changes look great to merge!

**Pull request merged**